### PR TITLE
Difficulty Bomb Delay to Summer 2022

### PR DESCRIPTION
**Pull request by q9f**
The average block times are increasing due to the difficulty bomb (also known as the "_ice age_") and slowly accelerating. This EIP proposes to delay the difficulty bomb for approximately ten months with the Vatnajoekull fork, a yet to be scheduled adjustment.

### CODE REVIEW

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:13]: This simply summary is a bit too long.  Think email subject line, GitHub pull request title, forum title, etc.

```suggestion
Delays the difficulty bomb so 30 second blocks won't happen until around summer 2022.
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:3]: Hard fork coordination happens separate from the EIP process in https://github.com/ethereum/eth1.0-specs/.  An EIP is just a technical document that describes a change, and has no concept of Ethereum mainnet fork names.

```suggestion
title: Difficulty Bomb Delay to Summer 2020
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:5]: `discussions-to` cannot be the PR itself.  A thread over at https://ethereum-magicians.org/ is preferred, but if you don't have or want to create an account there you can create a in issue in this repository for "Discussion of ..." and link to it here.

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:16]: As mentioned above, EIPs are unrelated to hardfork coordination.  While you are free to choose any variable names you want, I recommend something simpler and more clear given the lack of context of hard fork naming in this repository like `FORK_BLOCK_NUMBER`.  (on a personal note, I also recommend spelling out words since you'll write this document once and it will be read thousands of times)

Comment (MicahZoltu) [inline EIPS/eip-3238.md:16]: As mentioned above, try to keep the EIP purely technical.  Imagine someone is implementing this EIP for a chain that isn't Ethereum mainnet, which has no Homestead fork or Muir Glacier fork.  I also recommend against EIPs being *relative* changes, and instead be absolute changes.  So try to avoid statements like "+11M from previous".

```suggestion
Starting with `FORK_BLOCK_NUMBER` the client will calculate the difficulty based on a fake block number suggesting to the client that the difficulty bomb is adjusting eleven million blocks later than the actual block number.
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:31]: Same comment about keeping hardfork coordination out of the EIP (don't mention Berlin).

Also, you can be a bit more terse about needing a hard fork if you want (not required).  It is generally assumed that anyone reading this will understand what a hard fork is so you can just say something like "this change will require a hard fork" or "this is a consensus change, so it will require a hard fork".

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:33]: Recommend removing this whole paragraph as it is entirely hard fork coordination details (not appropriate for EIPs repository).
```suggestion
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:37]: Consider either removing this section entirely, or specifying some tests in this document.  This is a weak suggestion, and if you think there is significant value to developers to keep this mention I think that is fine, but I'm of the weak opinion that it doesn't add much value and shorter EIPs tend to be preferable over longer ones.

```suggestion
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:40]: Consider removing this section for the same reason as above.

If you decide to retain this section, I recommend removing the "should be implemented in all clients already" as this document should be written for future readers, which includes clients that may not be written at the time they are reading this EIP.

```suggestion
```

File: EIPS/eip-3238.md
Comment (MicahZoltu) [inline EIPS/eip-3238.md:41]: Security Considerations section is required.

```suggestion
## Security Considerations
There are no known security issues with this proposal.

## Copyright
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Title, discussions-to, and Security Considerations changes need to be made before merging as draft.  The rest I recommend considering and adjusting at this time as well as they will probably end up being blockers for moving this into Review status.

File: EIPS/eip-3238.md
Comment (q9f) [reply EIPS/eip-3238.md:3]: OK but fixed the year ;-)

```suggestion
title: Difficulty Bomb Delay to Summer 2022
```

File: EIPS/eip-3238.md
Comment (q9f) [reply EIPS/eip-3238.md:5]: OK

```suggestion
discussions-to: https://github.com/ethereum/EIPs/issues/3239
```

File: EIPS/eip-3238.md
Comment (q9f) [reply EIPS/eip-3238.md:31]: If it's not a blocker, I would like to keep these considerations in the proposal. I understand they should work "out of context," but we have to admit this is a strong ethereum mainnet context even when we just say "Difficulty Bomb Delay to Summer 2022."

Comment (MicahZoltu) [reply EIPS/eip-3238.md:31]: None of my comments on this section are a blocker for merging as a draft, but be warned that I probably won't merge a PR changing the status to Review as is.  Another editor might, but you'll have to shop around a bit.

We have been fighting a hard uphill battle getting hard fork coordination and governance stuff out of the EIP process and I am loath to give up this much ground in that battle, especially since it doesn't add any technical value to the EIP (it is purely governance discussion).

**Review by MicahZoltu — APPROVED** [review]
Will merge once CI passes.

**Pull request merged**
